lib/gce.sh

```diff
@@ -0,0 +1,40 @@
+# TODO generify this function to take a set of IPs
+# Set the internal and external ip arrays for the master node and all minions.
+# Sets: MASTER_IPS, MINION_NAMES, MINION_IPS, and HOSTS global vars.
+# Note: the MINION_IPS map is not yet referenced but remains for future usage.
+function gce_util::master_minions_ips() {
+	echo "-- Setting global master and minion internal and external ip addresses."
+	# format: (internal-ip external-ip)
+	MASTER_IPS=($(gcloud compute instances list --filter="zone:($GCP_ZONE) name:($GK_MASTER_NAME)" \
+		--format="value(networkInterfaces[0].networkIP,\
+		  networkInterfaces[0].accessConfigs[0].natIP)"))
+	if (( ${#MASTER_IPS[@]} == 0 )); then
+		echo "Failed to get master node $GK_MASTER_NAME's ip addresses."
+		return 1
+	fi
+
+	# Make a single call to gcloud to get all minion info
+	# format: (hostname internal-ip external-ip...)
+	local minions=($(gcloud compute instances list --filter="zone:($GCP_ZONE) name:($GK_NODE_NAME)" \
```

Comment:
ditto ^^
`util::get_instance_info <minion-filter> NAMES PRIVATE_IPS PUBLIC_IPS`

Reply:
Maybe... it would be useful if dealing with instances not in the default zone.

Reply:
A difference between aws and gce when writing a main script that needs to ssh to the target instances is for gce you need both the instance name and zone, but for aws the name along is sufficient. Therefore I've made the helper funs in _aws/ recognize the ZONES keyword (may be passed in by provider-agnostic caller)  but ignore it. My current thinking (at least for `util::get_instance_info` is that each provider needs to recognize the superset of keywords but set the value of keys that are not supported to "". Thoughts?

Reply:
We're not restricted to `gcloud compute ssh`, it's also possible to shell in directly with `ssh -i google_compute_engine.pub <external-ip>`, so it might be useful to right a generic function that does this for either provider.